### ISSUE TITLE
"no passphrase set." for already decrypted IdentityFile (KeyPairDeferred)

### ISSUE THREAD

**Issue by adnklauser** [mwiede/jsch#42]
I'm using this fork of jsch together with Spring Integration SFTP. The Spring Integration SFTP wrapper will do `setPassphrase` on an identity file ahead of time. That means I get a `KeyPairDeferred` where `encrypted` is set to `true`, but on its `delegate`, it's set to `false` (as it has been decrypted already by the earlier call to `setPassphrase`)

![image](https://user-images.githubusercontent.com/34234988/115748691-b3174280-a396-11eb-9ba0-641679f6e971.png)

Now `com.jcraft.jsch.UserAuthPublicKey#decryptKey` sees (correctly) that the identity is not encrypted and tries to call `setPassphrase(null)`
```java
if(!identity.isEncrypted() || passphrase!=null){
  if(identity.setPassphrase(passphrase)){
    if(passphrase!=null &&
       (session.getIdentityRepository() instanceof IdentityRepository.Wrapper)){
      ((IdentityRepository.Wrapper)session.getIdentityRepository()).check();
    }
    break;
  }
}
```

But then two calls down in `com.jcraft.jsch.KeyPair#decrypt(byte[])`, we check the `encrypted` field on the `KeyPairDeferred` itself instead, which means that the `decrypt` call doesn't abort early (already decrypted case), but instead falls into the "no passphrase set." case
```java
@Override
public boolean decrypt(byte[] _passphrase) {
    try {
        if (!encrypted) {
            return true;
        }
        if (_passphrase == null) {
            JSch.getLogger().log(Logger.ERROR, "no passphrase set.");
            return false;
        }
```

Shouldn't this be
```java
if(!isEncrypted()) {
```
instead for this scenario to work?

**Comment by mwiede** [comment]
Hi @adnklauser,
Thanks for reporting your issue.
So do I understand correct, you are using an encrypted Openssh-Format Key, then `Identity.setPassphrase` will decrypt it, and then later there is a problem with `KeyPairDeferred` ?
We have a junit Test `KeyPairIT`, which tests both cases, encrypted and unencrypted. Can you please provide a failing unit test for your issue?

**Comment by mwiede** [comment]
Thanks @chklauser I have now merged your testcase and adopted the code. The is no usage of com.jcraft.jsch.KeyPair#decrypt(java.lang.String) inside of the project, but I also overwrote it in KeyPairDeferred to be consistent.

**Comment by mwiede** [comment]
Fix is now released in 0.1.63
